src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Order.xml

```diff
@@ -187,6 +187,74 @@
                 </attribute>
             </itemOperation>
 
+            <itemOperation name="shop_account_change_payment_method">
+                <attribute name="method">PATCH</attribute>
+                <attribute name="path">/shop/account/orders/{id}/payments/{paymentId}</attribute>
+                <attribute name="messenger">input</attribute>
+                <attribute name="input">Sylius\Bundle\ApiBundle\Command\Account\ChangePaymentMethod</attribute>
+                <attribute name="denormalization_context">
+                    <attribute name="groups">shop:order:account:change_payment_method</attribute>
+                </attribute>
+                <attribute name="normalization_context">
+                    <attribute name="groups">shop:order:account:read</attribute>
+                </attribute>
+                <attribute name="openapi_context">
+                    <attribute name="summary">Change the payment method as logged shop user</attribute>
+                    <attribute name="parameters">
+                        <attribute>
+                            <attribute name="name">id</attribute>
+                            <attribute name="in">path</attribute>
+                            <attribute name="required">true</attribute>
+                            <attribute name="schema">
+                                <attribute name="type">string</attribute>
+                            </attribute>
+                        </attribute>
+                        <attribute>
+                            <attribute name="name">paymentId</attribute>
+                            <attribute name="in">path</attribute>
+                            <attribute name="required">true</attribute>
+                            <attribute name="schema">
+                                <attribute name="type">string</attribute>
+                            </attribute>
+                        </attribute>
+                    </attribute>
+                </attribute>
+            </itemOperation>
+
+            <itemOperation name="shop_guest_change_payment_method">
+                <attribute name="method">PATCH</attribute>
+                <attribute name="path">/shop/guest/orders/{id}/payments/{paymentId}</attribute>
```

Comment:
```suggestion
                <attribute name="path">/shop/guest/orders/{id}/payments/{paymentId}</attribute>
```
Does anyone have a better alternative for this route structure? This `guest` is a little bit confsing but I don't have better idea.